Project/Source/Components/ComponentFog.cpp

```diff
@@ -0,0 +1,77 @@
+#include "ComponentFog.h"
+
+#include "Application.h"
+#include "GameObject.h"
+#include "Modules/ModulePrograms.h"
+#include "Modules/ModuleRender.h"
+#include "Modules/ModuleEditor.h"
+#include "Modules/ModuleCamera.h"
+
+#include "GL/glew.h"
+#include "imgui.h"
+
+#define JSON_TAG_DENSITY "Density"
+#define JSON_TAG_FALLOFF "Falloff"
+#define JSON_TAG_INSCATTERING_COLOR "InscatteringColor"
+
+void ComponentFog::OnEditorUpdate() {
+	if (ImGui::Checkbox("Active", &active)) {
+		if (GetOwner().IsActive()) {
+			if (active) {
+				Enable();
+			} else {
+				Disable();
+			}
+		}
+	}
+	ImGui::Separator();
+	ImGui::DragFloat("Density", &density, App->editor->dragSpeed5f, 0.0f, inf, "%.4f");
+	ImGui::DragFloat("Falloff", &falloff, App->editor->dragSpeed5f, 0.0f, inf, "%.4f");
```

Comment:
I've seen the falloff at 0.5500 makes the fog completely invisible since I'm assuming it starts to flatten it out inversely? Have honestly no idea how that happens but it would be a good idea if there's a max value on Density and Falloff to limit it in the editor slider

Reply:
There is a bit of an artifaft when the height difference is too much, so I'm assuming that's what you saw. The thing is that it depends on both the falloff, the density and the Y difference, so it's not controllable.

Reply:
Yeah, thought it would be related, but I've tried tweaking some values and that seemed to be the limit for the falloff

Reply:
I can get a much higher falloff if I get the camera really close to the fog.